model_api/python/openvino/model_api/models/segmentation.py

```diff
@@ -41,10 +41,15 @@ def create_hard_prediction_from_soft_prediction(
     Returns:
         Numpy array of the hard prediction
     """
-    soft_prediction_blurred = cv2.blur(soft_prediction, (blur_strength, blur_strength))
-    assert len(soft_prediction.shape) == 3
-    soft_prediction_blurred[soft_prediction_blurred < soft_threshold] = 0
-    return np.argmax(soft_prediction_blurred, axis=2)
+    if blur_strength == -1 or soft_threshold == float("inf"):
```

Comment:
@Wovchena In the original code this if statement was flipped from what I can see. We discussed this in the past but the master implementation in c++ and python both use if (blur_strength == -1 and soft_threshold == infinity) else blur. But that should be an or based on the documentation. If either is soft_threshold or blur_strength are infinity or -1 (respectively) it should *not* blur.

I've applied this in this PR.